packages/odyssey-react-mui/src/labs/OdysseyPickers/SearchDropdown.tsx

```diff
@@ -0,0 +1,406 @@
+      bottom: 0,
+      alignSelf: "center",
+      width: odysseyDesignTokens.Spacing4,
+      height: "auto",
+      maxHeight: odysseyDesignTokens.Spacing4,
+      marginInlineEnd: odysseyDesignTokens.Spacing2,
+
+      svg: {
+        display: "flex",
+        width: "100%",
+        height: "auto",
+      },
+    }),
+  }),
+);
+
+type OptionAdornmentProps = {
+  adornment: Adornment;
+  adornmentSize: AdornmentSize;
+  odysseyDesignTokens: DesignTokens;
+};
+
+type ExtraProps = {
+  content: Adornment;
+  size?: AdornmentSize;
+  odysseyDesignTokens: DesignTokens;
+  onClick: () => void;
+};
+
+const CustomOptionContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{ odysseyDesignTokens: DesignTokens }>(() => ({
+  display: "flex",
+  justifyContent: "space-between",
+  width: "100%",
+}));
+
+const CustomOptionLeftContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{ odysseyDesignTokens: DesignTokens }>(() => ({
+  display: "flex",
+}));
+
+const OptionAdornment = ({
+  adornment,
+  adornmentSize,
+  odysseyDesignTokens,
+}: OptionAdornmentProps) => {
+  const isImageAdornment = typeof adornment === "string";
+
+  if (isImageAdornment) {
+    return (
+      <OptionAdornmentContainer
+        adornmentSize={adornmentSize}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        <img src={adornment} alt="" role="presentation" />
+      </OptionAdornmentContainer>
+    );
+  } else {
+    return (
+      <OptionAdornmentContainer
+        adornmentSize={adornmentSize}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        {adornment}
+      </OptionAdornmentContainer>
+    );
+  }
+};
+
+const Extra = ({ content, size, odysseyDesignTokens, onClick }: ExtraProps) => {
+  const handleClick = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    onClick();
+  };
+
+  return (
+    <OptionAdornmentContainer
+      adornmentSize={size}
+      odysseyDesignTokens={odysseyDesignTokens}
+      onClick={handleClick}
+    >
+      {content}
+    </OptionAdornmentContainer>
+  );
+};
+
+const CustomOption = <OptionType extends CustomOptionType>({
+  adornmentSize,
+  muiProps,
+  odysseyDesignTokens,
+  option,
+}: BaseOptionProps &
+  OptionProps<OptionType> & {
+    adornmentSize: AdornmentSize;
+  }) => {
+  const {
+    adornment,
+    description,
+    label,
+    metaData,
+    extra,
+    value,
+    onClick,
+    isInteractive = true,
+  } = option;
+
+  const handleOptionClick = (event: React.MouseEvent) => {
+    if (!isInteractive) {
+      event.preventDefault();
+      return;
+    }
+    event.stopPropagation();
+    onClick?.();
+  };
+
+  return (
+    <Option
+      hasAdornment
+      key={value}
+      muiProps={{
+        ...muiProps,
+        onClick: isInteractive ? handleOptionClick : undefined,
+        "aria-disabled": !isInteractive ? "true" : undefined,
+        role: "option",
+      }}
```

Comment:
Memoize this and any other you can find. These are the same comments from the other PR.